vyper/venom/passes/sccp/sccp.py

```diff
@@ -329,6 +338,40 @@ def _replace_constants(self, inst: IRInstruction):
                 if isinstance(lat, IRLiteral):
                     inst.operands[i] = lat
 
+    def _fix_phi_nodes(self):
+        visited: OrderedSet[IRBasicBlock] = OrderedSet()
+        for bb in self.cfg_phi_fix_needed:
+            self._fix_phi_bb_r(bb, visited)
+
+    def _fix_phi_bb_r(self, bb: IRBasicBlock, visited: OrderedSet[IRBasicBlock]):
+        if bb in visited:
+            return
+
+        for inst in bb.instructions:
+            if inst.opcode == "phi":
+                self._fix_phi_node_inst(inst)
+
+        visited.add(bb)
+
+        for next_bb in bb.cfg_out:
+            self._fix_phi_bb_r(next_bb, visited)
+
+    def _fix_phi_node_inst(self, phi_inst: IRInstruction):
+        if len(phi_inst.parent.cfg_in) != 1:
+            return
+
+        src_bb: IRBasicBlock = phi_inst.parent.cfg_in.first()
+        assert isinstance(src_bb, IRBasicBlock)
+
+        from_src_bb = filter(lambda x: x[0] == src_bb.label, phi_inst.phi_operands)
+        operands = list(map(lambda x: x[1], from_src_bb))
```

Comment:
please use a list comprehension here
```suggestion
        operands = [op for label, op in phi_inst.phi_operands if label == src_bb.label]
```